open-apparel-registry/open-apparel-registry
# Allow prefer contributor name in embed

## Overview

Adds a field to the embedded map config model allowing users to choose
to prefer their own contributed facility name over the default
facility name. In the EmbeddedMap, the facility name (not the address) of the facility
that was contributed by the embedded map's contributor is
promoted to the primary position. This is only be in effect in the
embedded map view, and not in the full site. If the contributor has
multiple active list items for the facility, we prefer first any that
match the assigned facility name, or secondly the most submission.

Connects #1550 

## Demo
<img width="484" alt="Screen Shot 2021-12-10 at 4 15 45 PM" src="https://user-images.githubusercontent.com/21046714/145643620-119d8deb-3461-4e22-8c99-a9ba34ef1fff.png">
<img width="609" alt="Screen Shot 2021-12-10 at 4 15 50 PM" src="https://user-images.githubusercontent.com/21046714/145643634-d59eec68-3795-417d-a776-09bb34309cb9.png">
<img width="308" alt="Screen Shot 2021-12-10 at 4 33 35 PM" src="https://user-images.githubusercontent.com/21046714/145644089-962527f5-d250-43ef-a718-8544e0264fcb.png">

## Notes

@lederer I adjusted the section hierarchy of the Embed settings page to add the new field checkbox as a separate subsection under a larger "fields" category, similar to the pattern in the Theme section. 

## Testing Instructions

* Create an embedded map config for a contributor.
* Run `./scripts/migrate`. Confirm the embed config you created before now has the field `prefer_contributor_name=false`
* Update a facility list item created by the contributor to have a different name from the associated facility
* View that facility in the facilities search and in its details page and confirm the name has not changed in the display
* Go to the embed config settings page for that contributor. The prefer contributor name checkbox should be false. 
* View the embed preview; the facility name in the list and details should be the same.
* Check the prefer contributor name box and the view the embed preview again. The facility name in the list and details should be the alternate name assigned to the list item. 

## Checklist
